kv/kvserver: TestClosedTimestampCantServeWithConflictingIntent failed

kv/kvserver.TestClosedTimestampCantServeWithConflictingIntent [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9682885?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9682885?buildTab=artifacts#/) on master @ [a4ab6b6243f509e62993b4d81855f23470d9901a](https://github.com/cockroachdb/cockroach/commits/a4ab6b6243f509e62993b4d81855f23470d9901a):


```
=== RUN   TestClosedTimestampCantServeWithConflictingIntent
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestClosedTimestampCantServeWithConflictingIntent1436482389
    test_log_scope.go:79: use -show-logs to present logs inline
    closed_timestamp_test.go:351: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/closed_timestamp_test.go:351
        	Error:      	Received unexpected error:
        	            	[NotLeaseHolderError] lease held by different store; r63: replica (n1,s1):1 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1681901373.794041544,0 exp=1681901379.793263093,0 pro=1681901373.793263093,0
        	Test:       	TestClosedTimestampCantServeWithConflictingIntent
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestClosedTimestampCantServeWithConflictingIntent1436482389
--- FAIL: TestClosedTimestampCantServeWithConflictingIntent (40.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100053 kv/kvserver: TestClosedTimestampCantServeWithConflictingIntent failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClosedTimestampCantServeWithConflictingIntent.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27142

kv/kvserver.TestClosedTimestampCantServeWithConflictingIntent [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9851509?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9851509?buildTab=artifacts#/) on master @ [3b1bc872bb4bf8e5eb2ad8790ba8ec781167dd1f](https://github.com/cockroachdb/cockroach/commits/3b1bc872bb4bf8e5eb2ad8790ba8ec781167dd1f):


```
=== RUN   TestClosedTimestampCantServeWithConflictingIntent
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestClosedTimestampCantServeWithConflictingIntent3629078817
    test_log_scope.go:79: use -show-logs to present logs inline
    closed_timestamp_test.go:337: request unexpectedly returned, should block; err: [NotLeaseHolderError] lease held by different store; r64: replica (n1,s1):1 not lease holder; current lease is repl=(n3,s3):2 seq=2 start=1682681331.525769256,0 exp=1682681337.524812851,0 pro=1682681331.524812851,0
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestClosedTimestampCantServeWithConflictingIntent3629078817
--- FAIL: TestClosedTimestampCantServeWithConflictingIntent (40.36s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100053 kv/kvserver: TestClosedTimestampCantServeWithConflictingIntent failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClosedTimestampCantServeWithConflictingIntent.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestClosedTimestampCantServeWithConflictingIntent [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9871089?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9871089?buildTab=artifacts#/) on master @ [4619767bb0b75ac85ddcb76e33c73211e369afed](https://github.com/cockroachdb/cockroach/commits/4619767bb0b75ac85ddcb76e33c73211e369afed):


```
=== RUN   TestClosedTimestampCantServeWithConflictingIntent
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestClosedTimestampCantServeWithConflictingIntent2101771649
    test_log_scope.go:79: use -show-logs to present logs inline
    closed_timestamp_test.go:351: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/closed_timestamp_test.go:351
        	Error:      	Received unexpected error:
        	            	[NotLeaseHolderError] lease held by different store; r64: replica (n1,s1):1 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1682767595.564162603,0 exp=1682767601.563077486,0 pro=1682767595.563077486,0
        	Test:       	TestClosedTimestampCantServeWithConflictingIntent
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestClosedTimestampCantServeWithConflictingIntent2101771649
--- FAIL: TestClosedTimestampCantServeWithConflictingIntent (40.83s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100053 kv/kvserver: TestClosedTimestampCantServeWithConflictingIntent failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClosedTimestampCantServeWithConflictingIntent.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestClosedTimestampCantServeWithConflictingIntent [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10123860?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10123860?buildTab=artifacts#/) on master @ [69875da00208dd792cb92aee54811e8a660773a9](https://github.com/cockroachdb/cockroach/commits/69875da00208dd792cb92aee54811e8a660773a9):


```
=== RUN   TestClosedTimestampCantServeWithConflictingIntent
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestClosedTimestampCantServeWithConflictingIntent464149897
    test_log_scope.go:79: use -show-logs to present logs inline
    closed_timestamp_test.go:351: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/closed_timestamp_test.go:351
        	Error:      	Received unexpected error:
        	            	[NotLeaseHolderError] lease held by different store; r64: replica (n1,s1):1 not lease holder; current lease is repl=(n3,s3):3 seq=2 start=1684314268.194372668,0 exp=1684314274.193422745,0 pro=1684314268.193422745,0
        	Test:       	TestClosedTimestampCantServeWithConflictingIntent
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestClosedTimestampCantServeWithConflictingIntent464149897
--- FAIL: TestClosedTimestampCantServeWithConflictingIntent (34.98s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100053 kv/kvserver: TestClosedTimestampCantServeWithConflictingIntent failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClosedTimestampCantServeWithConflictingIntent.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/closed_timestamp_test.go
+++ b/pkg/kv/kvserver/closed_timestamp_test.go
@@ -58,6 +58,11 @@ import (
 
 var aggressiveResolvedTimestampClusterArgs = base.TestClusterArgs{
 	ServerArgs: base.TestServerArgs{
+		RaftConfig: base.RaftConfig{
+			// With expiration-based leases, we may be unable to maintain leases under
+			// race and stress, so use a longer lease duration.
+			RangeLeaseDuration: time.Minute,
+		},
 		Knobs: base.TestingKnobs{
 			Store: aggressiveResolvedTimestampPushKnobs(),
 		},

